src/main/java/net/countercraft/movecraft/warfare/features/siege/tasks/SiegeProgressTask.java

```diff
@@ -39,8 +39,18 @@ public void run() {
             // Siege is done!
             endSiege();
             return;
-        } else {
-            // Siege is still in progress
+        }
+        // Siege is still in progress
+
+        // Check if it's time to begin sudden death
+        if (!siege.isSuddenDeathActive() && timeLeft < siege.getConfig().getSuddenDeathDuration()) {
+            Bukkit.getServer().broadcastMessage(String.format(I18nSupport.getInternationalisedString("Siege - Sudden Death"),
+                    siege.getName(), (timeLeft+2)/60));
+            if (!siege.leaderIsInControl()) {
+                endSiege();
```

Comment:
Won't this cause the siege to end twice since it's also ended in `SiegeLeaderListener`?

Reply:
No, because the SiegeLeaderListener one specifically fires when a craft is sunk or released, whereas the SiegeProgressTask one fires when sudden death begins and there is no flagship detected.